agent/api/task_test.go

```diff
@@ -997,6 +997,92 @@ func TestTaskUpdateKnownStatusChecksSteadyStateWhenSetToResourceProvisioned(t *t
 	assert.Equal(t, TaskRunning, testTask.GetKnownStatus())
 }
 
+// TestShouldWaitForExecutionCredentials tests the function ShouldWaitForExecutionCredentials
+func TestShouldWaitForExecutionCredentials(t *testing.T) {
+	testCases := []struct {
+		task   *Task
+		result bool
+		msg    string
+	}{
+		{&Task{
+			// Container configured to use execution role to pull and hasn't
+			// been pulled will need to wait for the credentials
+			KnownStatusUnsafe: TaskStatusNone,
+			Containers: []*Container{
+				{
+					RegistryAuthentication: &RegistryAuthenticationData{
+						Type: "ecr",
+						ECRAuthData: &ECRAuthData{
+							UseExecutionRole: true,
+						},
+					},
+					KnownStatusUnsafe: ContainerStatusNone,
+				},
+			},
+		}, true, "task needs credentials to pull container"},
+		{&Task{
+			// Container configured to not use execution role to pull and hasn't
+			// been pulled don't need to wait for the credentials
+			KnownStatusUnsafe: TaskStatusNone,
+			Containers: []*Container{
+				{
+					RegistryAuthentication: &RegistryAuthenticationData{
+						Type: "ecr",
+						ECRAuthData: &ECRAuthData{
+							UseExecutionRole: false,
+						},
+					},
+					KnownStatusUnsafe: ContainerStatusNone,
+				},
+			},
+		}, false, "no containe require credentials to pull, no need to wait for credentials"},
+		{&Task{
+			// Container configured to not use execution role to pull and has
```

Comment:
Container uses an execution role to pull, but the image has already been pulled, so there is no need to wait for credentials.